platform_umbrella/apps/common_ui/lib/common_ui/components/input_list.ex

```diff
@@ -107,28 +107,24 @@ defmodule CommonUI.Components.InputList do
     <div>
       <div :if={@label} class={label_class()}><%= @label %></div>
 
-      <%= if @field.value == [] do %>
-        <input type="hidden" name={"#{@field.name}[]"} />
-      <% else %>
-        <%= for {value, index} <- Enum.with_index(@field.value) do %>
-          <div class={wrapper_class()}>
-            <div class="flex-1">
-              <%= render_slot(
-                @inner_block,
-                Map.merge(@field, %{name: @field.name <> "[]", value: value})
-              ) %>
-            </div>
-
-            <.button
-              variant="minimal"
-              icon={:x_mark}
-              phx-value-index={index}
-              phx-click={@remove_click}
-              phx-target={@phx_target}
-            />
+      <.list :let={{value, index}} field={@field}>
+        <div class={wrapper_class()}>
+          <div class="flex-1">
+            <%= render_slot(
+              @inner_block,
+              Map.merge(@field, %{name: @field.name <> "[]", value: value})
+            ) %>
           </div>
-        <% end %>
-      <% end %>
+
+          <.button
+            variant="minimal"
+            icon={:x_mark}
+            phx-value-index={index}
+            phx-click={@remove_click}
+            phx-target={@phx_target}
+          />
```

Comment:
If we use the drop param names this might be able to delete the last one?

Reply:
See https://github.com/batteries-included/batteries-included/issues/904#issuecomment-2369790876